src/color.rs

```diff
@@ -91,51 +144,280 @@ impl Serialize for RGBA {
+
+// NOTE: LCH and OKLCH is not declared inside the [impl_lch_like] macro,
+// because it causes cbindgen to ignore them.
+
+/// Color specified by lightness, chroma and hue components.
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[repr(C)]
+pub struct Lch {
+    /// The lightness component.
+    pub lightness: f32,
+    /// The chroma component.
+    pub chroma: f32,
+    /// The hue component.
+    pub hue: f32,
+    /// The alpha component.
+    pub alpha: f32,
+}
+
+/// Color specified by lightness, chroma and hue components.
+#[derive(Clone, Copy, PartialEq, Debug)]
+#[repr(C)]
+pub struct Oklch {
+    /// The lightness component.
+    pub lightness: f32,
+    /// The chroma component.
+    pub chroma: f32,
+    /// The hue component.
+    pub hue: f32,
+    /// The alpha component.
+    pub alpha: f32,
+}
+
+macro_rules! impl_lch_like {
+    ($cls:ident, $fname:literal) => {
+        impl $cls {
+            /// Construct a new color with lightness, chroma and hue components.
+            pub fn new(lightness: f32, chroma: f32, hue: f32, alpha: f32) -> Self {
+                Self {
+                    lightness,
+                    chroma,
+                    hue,
+                    alpha,
+                }
+            }
+        }
+
+        #[cfg(feature = "serde")]
+        impl Serialize for $cls {
+            fn serialize<S>(&self, serializer: S) -> Result<serde::ser::Ok, serde::ser::Error>
+            where
+                S: Serializer,
+            {
+                (self.lightness, self.chroma, self.hue, self.alpha).serialize(serializer)
+            }
+        }
+
+        #[cfg(feature = "serde")]
+        impl<'de> Deserialize<'de> for $cls {
+            fn deserialize<D>(deserializer: D) -> Result<Self, serde::de::Error>
+            where
+                D: Deserializer<'de>,
+            {
+                let (lightness, chroma, hue, alpha) = Deserialize::deserialize(deserializer)?;
+                Ok(Self::new(lightness, chroma, hue, alpha))
+            }
+        }
+
+        impl ToCss for $cls {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result
+            where
+                W: fmt::Write,
+            {
+                dest.write_str($fname)?;
+                dest.write_str("(")?;
+                self.lightness.to_css(dest)?;
+                dest.write_char(' ')?;
+                self.chroma.to_css(dest)?;
+                dest.write_char(' ')?;
+                self.hue.to_css(dest)?;
+                serialize_alpha(dest, self.alpha, false)?;
+                dest.write_char(')')
+            }
+        }
+    };
+}
+
+impl_lch_like!(Lch, "lch");
+impl_lch_like!(Oklch, "oklch");
+
+/// An absolutely specified color.
+/// https://w3c.github.io/csswg-drafts/css-color-4/#typedef-absolute-color-base
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum AbsoluteColor {
+    /// Specify sRGB colors directly by their red/green/blue/alpha chanels.
+    RGBA(Rgba),
```

Comment:
Actually I meant the variant names should be `Rgba()` etc (so I was expecting `Rgba(RGBA)`, sorry if that wasn't clear). But `Rgba(Rgba)` is also alright.

Reply:
And now those are renamed too :smile: